gap/Derivations.gd

```diff
@@ -1,63 +1,367 @@
+#############################################################################
+##
+##                                               CAP package
+##
+##  Copyright 2013, Sebastian Gutsche, TU Kaiserslautern
+##                  Sebastian Posur,   RWTH Aachen
+##
+#! @Chapter Managing derived methods
+#!
+##
+#############################################################################
+
+
+####################################
+##
+#! @Section Info class
+##
+####################################
+
+#! @Description
+#!  Info class for derivations.
 DeclareInfoClass( "DerivationInfo" );
+
+#!
 DeclareGlobalFunction( "ActivateDerivationInfo" );
+
+#!
 DeclareGlobalFunction( "DeactivateDerivationInfo" );
 
+
+####################################
+##
+#! @Section Derivation objects
+##
+####################################
+
+#! @Description
+#!  A derivation object describes a derived method.
+#!  It contains information about which operation the derived method
+#!  implements, and which other operations it relies on.
 DeclareCategory( "IsDerivation", IsObject );
```

Comment:
- IsDerivedMethod would be OK.
- Mathematical properties should never be hijacked as categories: Imagine one defines something more general (e.g., an additive homomorphism) and wants to test if it is a derivation w.r.t. an underlying multiplicative structure. If IsDerivation is a GAP category one would be stuck. Unfortunately in the extremely awful Bourbaki/set-theory style building of traditional GAP objects things like IsGroup is a GAP category :(
- Sorry, I just saw IsDerivation in your PR, I couldn't object earlier :)